src/components/ColorPicker.tsx

```diff
@@ -237,15 +238,31 @@ export const ColorPicker = ({
   color,
   onChange,
   label,
+  shortcut,
 }: {
   type: "canvasBackground" | "elementBackground" | "elementStroke";
   color: string | null;
   onChange: (color: string) => void;
   label: string;
+  shortcut?: string;
 }) => {
   const [isActive, setActive] = React.useState(false);
   const pickerButton = React.useRef<HTMLButtonElement>(null);
 
+  const handleKeyDown = (event: KeyboardEvent) => {
+    if (!isActive && event.key === shortcut) {
+      setActive(true);
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener(EVENT.KEYDOWN, handleKeyDown, false);
+
+    return () => {
+      document.removeEventListener(EVENT.KEYDOWN, handleKeyDown, false);
+    };
+  });
```

Comment:
Doesn't this create global event handlers for each ColorPicker instance? In our case that would be 2 event handlers. One of which would be bound always (canvas picker), and the 2nd would be bound/unbound when toggling the element properties menu. Probably both is ok from perf perspective, but longer-term we want to route all keyboard handling through one handler (to enable host apps to have control).

So I'd reuse the app's keydown handler and invoke the color picker from there. Problem is on the "invoke". We might have to make the color pickers controlled and use appState for the toggle states.

Alternatively, we can expose a hook for our app keydown handler which the color pickers would subscribe to (though that would triple the amount of checks per keystroke).

Other ideas welcome.

Reply:
I'll use the app's keydown handler and do this. If I come up with any other ideas along the way, I'll post it here.

Also, I'm having trouble understanding how we can expose a hook for our app keydown handler? Are you suggesting to create a hook that exposes the keydown handler used in app now? If yes, how? since App.tsx is a class component. 🤔